test/app/presenter/ChessBoardPresenter.spec.ts

```diff
@@ -1,26 +1,61 @@
 import '@testing-library/jest-dom';
 import { ChessBoardView } from '../../../src/app/view/ChessBoardView';
-import { ChessModel } from '../../../src/app/model/ChessModel';
+import { ChessModel, PIECES_START_POSITION, Square } from '../../../src/app/model';
 import { ChessBoardPresenter } from '../../../src/app/presenter/ChessBoardPresenter';
 import { ViewEventBus } from '../../../src/app/view/events/ViewEventBus';
 import { ViewEvent } from '../../../src/app/view/events/ViewEvent';
 import { SquareWasClicked } from '../../../src/app/view/events/SquareWasClicked';
 import { InMemoryViewEventBus } from '../../../src/app/view/events/InMemoryViewEventBus';
-import { PIECES_START_POSITION } from '../../../src/app/model/Constances';
 
 describe('ChessBoardPresenter', () => {
-  const viewEvents: ViewEventBus = new InMemoryViewEventBus();
-  const view: ChessBoardView = chessBoardViewMock(viewEvents);
-  const model: ChessModel = { squaresWithPiece: PIECES_START_POSITION, move: jest.fn() };
-  const presenter: ChessBoardPresenter = new ChessBoardPresenter(view, model);
+  it('when square A1 was clicked on the view, then selected piece should be shown with its possible moves', () => {
+    const viewEvents: ViewEventBus = new InMemoryViewEventBus();
+    const view: ChessBoardView = chessBoardViewMock(viewEvents);
+    const model: ChessModel = {
+      squaresWithPiece: PIECES_START_POSITION,
+      move: jest.fn(),
+      possibleMoves(position: Square): Square[] {
+        return [];
+      },
+    };
+    new ChessBoardPresenter(view, model);
 
-  it('when square was clicked on the view, then selected piece should be shown', () => {
     viewEvents.publish(new SquareWasClicked({ x: 1, y: 1 }));
 
-    expect(view.showSelectedPiece).toHaveBeenCalledWith({ x: 1, y: 1 });
+    expect(view.showSelectedPiece).toHaveBeenCalledWith('a1');
+    expect(view.hideSelection).toHaveBeenCalled();
+    expect(view.hideAllAvailableMoves).toHaveBeenCalled();
+    expect(view.showAvailableMoves).toHaveBeenCalledWith([]);
+  });
+
+  it('when square A2 was clicked on the view, then selected piece should be shown with its possible moves', () => {
+    const viewEvents: ViewEventBus = new InMemoryViewEventBus();
+    const view: ChessBoardView = chessBoardViewMock(viewEvents);
+    const model: ChessModel = {
+      squaresWithPiece: PIECES_START_POSITION,
+      move: jest.fn(),
+      possibleMoves(position: Square): Square[] {
+        return [
+          { column: 'A', row: 3 },
+          { column: 'A', row: 4 },
+        ];
```

Comment:
@tomdworniczak że niby tak tutaj trzeba było zrobić to possibleMoves, żeby dobrze `expect(view.showAvailableMoves).toHaveBeenCalledWith(['a3', 'a4']);` to działało... sprytnie, ale w końcu co z tym co wczoraj ewaluowaliśmy? Bo przecież w metodzie possibleMoves w ChessEngine 
`possibleMoves(position: Square): Square[] {
    return this.board.onPositionPiece(position)?.possibleMoves(position, this.board) ?? [];
  }`
  
 to `this.board.onPositionPiece(position)` zwracało nam undefined. To w końcu to dobrze jest? Niby działa ok, ale czemu nam tak wyrzucało wczoraj...wtf?